platforms/fabric/src/main/java/com/dfsek/terra/fabric/mixin/lifecycle/server/ServerMainMixin.java

```diff
@@ -29,13 +38,31 @@
 
 @Mixin(Main.class)
 public class ServerMainMixin {
+    @Shadow
+    @Final
+    private static Logger LOGGER;
+    
     @Inject(method = "main([Ljava/lang/String;)V",
             at = @At(value = "INVOKE",
-                     target = "Lnet/minecraft/util/registry/DynamicRegistryManager;create()" +
-                              "Lnet/minecraft/util/registry/DynamicRegistryManager$Impl;"))
+                     target = "Lnet/minecraft/server/SaveLoader;refresh()V") // after registry manager creation
+            )
     private static void injectConstructor(String[] args, CallbackInfo ci) {
         FabricEntryPoint.getPlatform().getEventManager().callEvent(
                 new PlatformInitializationEvent()); // Load during MinecraftServer construction, after other mods have registered blocks
         // and stuff
     }
+    
+    @Redirect(method = "method_40373(Lnet/minecraft/world/level/storage/LevelStorage$Session;Ljoptsimple/OptionSet;" +
+                       "Ljoptsimple/OptionSpec;Lnet/minecraft/server/dedicated/ServerPropertiesLoader;Ljoptsimple/OptionSpec;" +
+                       "Lnet/minecraft/resource/ResourceManager;Lnet/minecraft/resource/DataPackSettings;)" +
+                       "Lcom/mojang/datafixers/util/Pair;",
+              at = @At(value = "INVOKE",
+                       target = "net/minecraft/util/registry/DynamicRegistryManager.createAndLoad ()" +
+                                "Lnet/minecraft/util/registry/DynamicRegistryManager$Mutable;"))
+    private static Mutable injectBiomes() {
+        Mutable mutable = DynamicRegistryManager.createAndLoad();
+        LOGGER.info("Injecting Terra biomes...");
+        FabricUtil.registerBiomes(mutable.get(Registry.BIOME_KEY));
```

Comment:
Should this not dispatch the `BiomeRegistrationEvent` event?

Reply:
No.